code/parachain/runtime/dali/src/lib.rs

```diff
@@ -1367,12 +1368,34 @@ impl_runtime_apis! {
 			SafeRpcWrapper(<Assets as fungibles::Inspect::<AccountId>>::balance(asset_id, &account_id))
 		}
 
-		fn list_assets() -> Vec<Asset<ForeignAssetId>> {
-			let mut assets = CurrencyId::list_assets();
-			let mut foreign_assets = assets_registry::Pallet::<Runtime>::get_foreign_assets_list();
-			assets.append(&mut foreign_assets);
-
-			assets
+		fn list_assets() -> Vec<Asset<Balance, ForeignAssetId>> {
+			// Hardcoded assets
+			let assets = CurrencyId::list_assets().into_iter().map(|mut asset| {
+				// Add hardcoded ratio and ED for well known assets
+				asset.ratio = WellKnownForeignToNativePriceConverter::get_ratio(CurrencyId(asset.id));
+				asset.existential_deposit = multi_existential_deposits::<AssetsRegistry>(&asset.id.into());
+				asset
+			}).collect::<Vec<_>>();
+			// Assets from the assets-registry pallet
+			let foreign_assets = assets_registry::Pallet::<Runtime>::get_foreign_assets_list();
```

Comment:
This will serve also in the case we update anything from a hardcoded assets via `assets-registry`. If we don't add nor update, this will be empty and have no effect on the RPC response.